spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkCatalogCacheExpiration.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import java.util.Map;
+import org.apache.iceberg.CachingCatalog;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.spark.SparkCatalog;
+import org.apache.iceberg.spark.SparkCatalogTestBase;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestSparkCatalogCacheExpiration extends SparkCatalogTestBase {
```

Comment:
I don't think that this should inherit from `SparkCatalogTestBase`. We don't need to set up 4 different Spark catalogs for this. Can you use `SparkTestBaseWithCatalog` instead?

Reply:
I actually override it with 4 catalogs to test different combinations of behaviors.

I can change the base, but the outcome will still be the same.

Let me re-evaluate the test cases now that some changes have been made.

Reply:
I guess I can, but all of these things test different things. Also, `SparkTestBaseWIthCatalog` errors out when I try to pass it any config (which I would need here).

Reply:
Ok. So I figured out how to make `SparkTestBaseWithCatalog` work. And I used it to test the SparkSessionCatalog.

But I'm not sure which test case I should get rid of here:
1) cache-enabled = false
2) cache-enabled = true, expiration interval is above zero
3) cache-enabled = true, expiration interval is zero so caching shouldn't happen.

I feel I can get rid of test case number one just fine, but I'd like to keep the other two.

Reply:
Ok. So in the interest of speed / being able to parallelize test suites, as well as making the code cleaner, I've left this as only two test case.

This file has two test cases, one for cache enabled / exipration enabled and one for cache enabled / expiration disabled.

I added a separate test using `SparkTestBaseWithCatalog` that tests implicitly turning off caching by using a value of 0 for the expiration interval.

I also added a separate test using `SparkTestBaseWithCatalog` that tests one example of cache expiration enabled on the SparkSessionCatalog.

Reply:
So there are now 4 tests. There are a lot of possible combinations to test, let me know which ones you think I should remove.

1. TestSparkCatalogCacheExpirationDisabled
  1a. Checks that CachingCatalog is not used  when cache.expiration-interval-millis=0 (this one seems important).
2. TestSparkCatalogCacheExpirationEnabled
  2a. Check with cache-enabled and expiration enabled via 30s (in millis)
  2b. Check with cache-enabled and expiration disabled via `-1`
3. TestSparkSessionCatalogCacheExpiration
  3a. Same as 2a, but using the SparkSessionCatalog

I think that 1a is important to have a test that verifies that behavior, same with 2b. If we wanted to get rid of a test, I would vote for 2a and just use 3a instead (since that tests the same behavior but ALSO ensures that the SparkSessionCatalog works with the new configs as well).

Reply:
The reason that we need to setup different Spark catalogs to test this is that the configuration that gets passed into `initialize`, so a second behavior test can't be run over the same catalog.

Again, let me know which tests to remove. And when I backport it, I'll only backport a subset of the tests if we leave multiple in this one. But I think at least 2 of these are necessary checks (`cache.expiration-interval-millis=0` disabling caching entirely and `cache.expiration-interval-millis=-1` for cache forever). Ideally we'd leave 3 but I understand these Spark tests are expensive.

Reply:
I managed to get them into one test class that only instantiates one spark session.